runtime/exporter-otlp/src/main/java/io/aklivity/zilla/runtime/exporter/otlp/internal/config/OtlpExporterConfig.java

```diff
@@ -60,8 +60,39 @@ public OtlpExporterConfig(
         OtlpEndpointConfig endpoint = options.endpoint;
         URI location = endpoint.location;
         OtlpOverridesConfig overrides = endpoint.overrides;
-        this.metrics = location.resolve(overrides.metrics);
-        this.logs = location.resolve(overrides.logs);
+
+        this.metrics = normalizeAndJoinPaths(location, overrides.metrics);
+        this.logs = normalizeAndJoinPaths(location, overrides.logs);
+    }
+
+    private URI normalizeAndJoinPaths(URI base, String path) {
+        String basePath = base.getPath();
+        if (basePath == null) {
+            basePath = "/";
+        }
+        
+        String pathToAppend = path;
+        if (pathToAppend.startsWith("/")) {
+            pathToAppend = pathToAppend.substring(1);
+        }
+        
+        if (!basePath.endsWith("/")) {
+            basePath += "/";
+        }
+        
+        try {
+            return new URI(
+                base.getScheme(),
+                base.getUserInfo(),
+                base.getHost(),
+                base.getPort(),
+                basePath + pathToAppend,
+                base.getQuery(),
+                base.getFragment()
+            );
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to join URI paths", e);
+        }
```

Comment:
This approach loses the ability to fully override the endpoint with a different URI that includes scheme host and port as well as path.


```suggestion

        this.metrics = resolveOverride(location, overrides.metrics);
        this.logs = resolveOverride(location, overrides.logs);
    }

    private URI resolveOverride(
        URI location,
        String override)
    {
        URI normalized = Optional.ofNullable(location.getPath())
            .map(p -> p.endsWith("/") ? p : p + "/")
            .map(location::resolve)
            .orElse(location.resolve("/"));

        URI overrideURI = Optional.of(URI.create(override))
           .map(...)
        // TODO: check for missing authority to treat as path only
        // TODO: if path has leading / then remove
           .get();

        // TODO: resolve overrideURI against normalized location to support absolute or relative overrides
    }
```